File: Codeforce/2023/RPC12/King.py
import math

cantidad = int(input())
puntos = []

for i in range(cantidad):
    x, y = map(int, input().split())
    puntos.append((x, y))

distancia = 999999999

for puntoPivote in puntos:
    sumaDistancias = 0

    for puntoAlejados in puntos:
        sumaDistancias += math.sqrt((puntoPivote[0] - puntoAlejados[0]) ** 2 + (puntoPivote[1] - puntoAlejados[1]) ** 2)

    distancia = min(distancia, sumaDistancias / (cantidad - 1))

print("{:.9f}".format(distancia))

File: Codeforce/2023/RPC12/King.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#include <iomanip>
using namespace std;

int main() {
    float cantidad;
    cin >> cantidad;
    vector<pair<int, int>> puntos;

    for (int i = 0; i < cantidad; i++) {
        int x, y;
        cin >> x >> y;
        puntos.push_back(make_pair(x, y));
    }

    float distancia = 999999999;
    
    for (auto puntoPivote : puntos) {
        float sumaDistancias = 0;
        
        for (auto puntoAlejados : puntos) {
            sumaDistancias += sqrt(pow(puntoPivote.first - puntoAlejados.first, 2) + pow(puntoPivote.second - puntoAlejados.second, 2));
        }
        
        distancia = min(distancia, sumaDistancias / (cantidad - 1));
    }

    cout << fixed << std::setprecision(9) << distancia << endl; // Usar 'fixed' para mostrar los decimales
}
